### PR TITLE
Major Major Refactor

### PR DESCRIPTION
**Pull request by NeilBaner**
The code was a massive pile of gobbledegook, because many people wrote it adopting many different styles. I'm making an attempt to clean it up so it's not only better to look at, but also easier to work with.

### CODE REVIEW

**Comment by NeilBaner** [comment]
I'll try to finish as much as I can by 2100 today, after which you guys can review, merge and work on it further. Please do try to follow the style, and OOP conventions, I used as much as possible for the sake of consistency. Thanks all!

File: src/main/java/seedu/duke/userinterface/command/notebook/AddNotebookMode.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            break;
+        case NOTEBOOK_SECTION:
+            currentSection.addPage(title, content);
+            break;
+        default:
+            System.out.println("\tunable to add notebook/section/page");
+            break;
+        }
+    }
+}
```
Comment (franceneee) [inline src/main/java/seedu/duke/userinterface/command/notebook/AddNotebookMode.java]: Sooooooo much neater omg, thank you.

File: src/main/java/seedu/duke/userinterface/InputParser.java [hunk 2]
```diff
@@ -7,27 +7,25 @@
...
-import seedu.duke.userinterface.command.Remove;
-import seedu.duke.userinterface.command.RemoveTask;
-import seedu.duke.userinterface.command.Select;
+import seedu.duke.userinterface.command.notebook.AddNotebookMode;
+import seedu.duke.userinterface.command.notebook.ListNotebookMode;
+import seedu.duke.userinterface.command.notebook.RemoveNotebookMode;
+import seedu.duke.userinterface.command.notebook.SelectNotebookMode;
+import seedu.duke.userinterface.command.timetable.AddTimetableMode;
+import seedu.duke.userinterface.command.timetable.ListTimetableMode;
+import seedu.duke.userinterface.command.timetable.RemoveTimetableMode;
```
Comment (franceneee) [inline src/main/java/seedu/duke/userinterface/InputParser.java]: Good effort in making this section neater xD

File: src/main/java/seedu/duke/userinterface/InputParser.java [hunk 3]
```diff
@@ -7,27 +7,25 @@
...
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
 
-import static seedu.duke.userinterface.command.AddTimetable.DEADLINE_DELIMITER;
-import static seedu.duke.userinterface.command.AddTimetable.TASK_DELIMITER;
-import static seedu.duke.userinterface.command.Select.NOTEBOOK_DELIMITER;
-import static seedu.duke.userinterface.command.Select.PAGE_DELIMITER;
-import static seedu.duke.userinterface.command.Select.SECTION_DELIMITER;
+import static seedu.duke.userinterface.command.notebook.SelectNotebookMode.*;
```
Comment (franceneee) [inline src/main/java/seedu/duke/userinterface/InputParser.java]: Probably should not have wildcard imports...

**Review by franceneee — APPROVED** [review]
Great job! Everything is so much neater now!

File: src/main/java/seedu/duke/userinterface/InputParser.java (hunk 2 above)
Comment (NeilBaner) [reply src/main/java/seedu/duke/userinterface/InputParser.java]: Thanks, it was ~~one press of `Ctrl-Alt-L`~~ a lot of hard work and software engineering, yes, that's definitely what that was...

File: src/main/java/seedu/duke/userinterface/InputParser.java (hunk 3 above)
Comment (NeilBaner) [reply src/main/java/seedu/duke/userinterface/InputParser.java]: I blame IntelliJ for automatically "optimising" that

File: src/main/java/seedu/duke/userinterface/command/notebook/AddNotebookMode.java (hunk 1 above)
Comment (NeilBaner) [reply src/main/java/seedu/duke/userinterface/command/notebook/AddNotebookMode.java]: :)

**Comment by NeilBaner** [comment]
We really need to rewrite the IO redirection tests...

**Review by yAOwzers — APPROVED** [review]
Ok overall looks good to merge

**Pull request merged**